src/cmd.ts

```diff
@@ -30,38 +29,42 @@ import { doStartupValidation } from "./startup.js";
 import { parseTorrentFromFilename } from "./torrent.js";
 import { fallback } from "./utils.js";
 
-function processOptions(options): RuntimeConfig {
-	if (options.rssCadence) {
-		options.rssCadence = Math.max(ms(options.rssCadence), ms("10 minutes"));
-	}
-	if (options.searchCadence) {
-		options.searchCadence = ms(options.searchCadence);
+const fileConfig = await getFileConfig();
 
-		// Set a minimum cadence of once a day if excludeRecentSearch hasn't been set to avoid flooding requests
-		if (!options.excludeRecentSearch) {
-			options.searchCadence = Math.max(
-				options.searchCadence,
-				ms("1 day")
-			);
-		}
-	}
+/**
+ * validates and sets RuntimeConfig
+ * @return (the number of errors Zod encountered in the configuration)
+ */
 
-	if (options.excludeOlder) {
-		options.excludeOlder = ms(options.excludeOlder);
-	}
-	if (options.excludeRecentSearch) {
-		options.excludeRecentSearch = ms(options.excludeRecentSearch);
-	}
-	if (options.snatchTimeout) {
-		options.snatchTimeout = ms(options.snatchTimeout);
-	}
-	if (options.searchTimeout) {
-		options.searchTimeout = ms(options.searchTimeout);
+export async function validateAndSetRuntimeConfig(options: RuntimeConfig) {
+	let zodErrorCount = 0;
+	options.configValid = true;
+	setRuntimeConfig(options);
+	initializeLogger();
+	logger.info(`${PROGRAM_NAME} v${PROGRAM_VERSION}`);
+	logger.info("Validating your configuration...");
+	try {
+		options = VALIDATION_SCHEMA.parse(await getFileConfig(), {
```

Comment:
This was a remanent from when processOptions was still present and I was debugging the ms conversion function, since processOptions would attempt to convert and leave it undefined if invalid I was left to use fileConfig instead to work out that function or would have had to remove/modify processOptions then - it was just easier to parse the config during testing/development.

Since we've removed processOptions now, I think we can go to validation of the options variable instead now.